### PR TITLE
feat/DP-1433 | reduce the minimum commit.size.byte value to 100KB

### PR DESCRIPTION
**Pull request by afiore**
## Highlights:

- reduce the minimum acceptable `commit.size.bytes` to 100Kb (rather 1MB).
- reduce the default `row.group.size.bytes` to 10 KB (rather than 1MB).
- tweak `PropertiesHelper.getInt` for easier testing.

## Testing 
- added basic unit test coverage.
- Introduce an e2e test verifying the following invariant:
 file size <= `commit.size.bytes` + `row.group.size.bytes`

### CODE REVIEW

File: connector/src/main/scala/com/celonis/kafka/connect/ems/config/EmsSinkConfigConstants.scala [hunk 1]
```diff
@@ -57,7 +57,7 @@ object EmsSinkConfigConstants {
     s"The time interval in milliseconds to upload the data to EMS if the other two commit policies are not yet applicable."
 
   val PARQUET_ROW_GROUP_SIZE_BYTES_KEY:     String = s"$CONNECTOR_PREFIX.parquet.row.group.size.bytes"
-  val PARQUET_ROW_GROUP_SIZE_BYTES_DEFAULT: Int    = 1024 * 1024
+  val PARQUET_ROW_GROUP_SIZE_BYTES_DEFAULT: Int    = 10 * 1024 // 1024
```
Comment (kowallen) [inline connector/src/main/scala/com/celonis/kafka/connect/ems/config/EmsSinkConfigConstants.scala:60]: unnecessary comment

**Review by kowallen — APPROVED** [review]

File: connector/src/main/scala/com/celonis/kafka/connect/ems/config/ParquetConfig.scala [hunk 2]
```diff
@@ -38,12 +39,18 @@ object ParquetConfig {
...
           error(PARQUET_ROW_GROUP_SIZE_BYTES_KEY, "The parquet row group size must be at least 1.")
-        else value.asRight[String]
+        else if (value > commitPolicyFileSize) {
+          logger.warn(
+            s"Supplied value for $PARQUET_ROW_GROUP_SIZE_BYTES_KEY conflicts with $COMMIT_SIZE_KEY (i.e. $value > $commitPolicyFileSize ). Overriding to $commitPolicyFileSize",
+          )
+          commitPolicyFileSize.toInt.asRight
+        } else value.asRight[String]
+      }
       case None => PARQUET_ROW_GROUP_SIZE_BYTES_DEFAULT.asRight
```
Comment (nicmart) [inline connector/src/main/scala/com/celonis/kafka/connect/ems/config/ParquetConfig.scala:54]: The default may bypass validation, if `parquet.row.group.size.bytes` is not provided and `commit.size.bytes` is smaller than that default value `10_000`.

File: connector/src/main/scala/com/celonis/kafka/connect/ems/config/ParquetConfig.scala [hunk 3]
```diff
@@ -38,12 +39,18 @@ object ParquetConfig {
...
         if (value < 1)
           error(PARQUET_ROW_GROUP_SIZE_BYTES_KEY, "The parquet row group size must be at least 1.")
-        else value.asRight[String]
+        else if (value > commitPolicyFileSize) {
+          logger.warn(
+            s"Supplied value for $PARQUET_ROW_GROUP_SIZE_BYTES_KEY conflicts with $COMMIT_SIZE_KEY (i.e. $value > $commitPolicyFileSize ). Overriding to $commitPolicyFileSize",
+          )
+          commitPolicyFileSize.toInt.asRight
+        } else value.asRight[String]
+      }
```
Comment (nicmart) [inline connector/src/main/scala/com/celonis/kafka/connect/ems/config/ParquetConfig.scala:53]: The default may bypass validation, if `parquet.row.group.size.bytes` is not provided and `commit.size.bytes` is smaller than that default value `10_000`.

Comment (afiore) [reply connector/src/main/scala/com/celonis/kafka/connect/ems/config/ParquetConfig.scala:53]: You nearly convinced me there, but [we do validate](https://github.com/celonis/kafka-ems-connector/blob/46ac9079734c536a7dced5430a980d0afb185f35/connector/src/main/scala/com/celonis/kafka/connect/ems/config/CommitPolicyConfig.scala#L28) that `commit.size.bytes` is > 10_000. Or am I missing something?

**Pull request merged**